base/deprecated.jl

```diff
@@ -1591,6 +1591,11 @@ end
 @deprecate readstring(filename::AbstractString) read(filename, String)
 @deprecate readstring(cmd::AbstractCmd) read(cmd, String)
 
+@deprecate +(a::Number, b::AbstractArray) a .+ b
+@deprecate +(a::AbstractArray, b::Number) a .+ b
+@deprecate -(a::Number, b::AbstractArray) a .- b
+@deprecate -(a::AbstractArray, b::Number) a .- b
```

Comment:
Could you add a note here indicating that when this deprecations get removed, we should rewrite this methods throwing a `MethodError` that suggests using the dot version and explaining why it fails? (I imagine this could be regular pitfall for a lot of people)